starboard/shared/x11/application_x11.cc

```diff
@@ -861,13 +858,16 @@ void ApplicationX11::PlayerSetBounds(SbPlayer player,
   frame_info.width = width;
   frame_info.height = height;
 
-  if (player_exists) {
-    return;
+  // The bounds should only take effect once the UI frame is submitted. But we
+  // also apply the bounds immediately so that there is no flicker.
+  for (auto it = current_video_bounds_.begin();
+       it != current_video_bounds_.end(); ++it) {
+    if (it->player == player) {
+      current_video_bounds_.erase(it);
+      break;
+    }
   }
```

Comment:
Chromium supports most of C++20 and in particular 
std::vector::erase_if() ([here](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#std_erase_if_for-containers-allowed)) so we could just do here
```
std::erase_if(current_video_bounds_, [player](const auto& entry) {
    return entry.player == player;
});
```

Albeit I'm not sure what is supported by Starboard-Linux. Maybe the
same?